staking/programs/staking/src/state/vesting.rs

```diff
@@ -233,15 +233,80 @@ impl VestingSchedule {
             amount,
         }))
     }
+
+    pub fn split_vesting_schedule(
+        &self,
+        transferred_amount: u64,
+        total_amount: u64,
+    ) -> Result<(VestingSchedule, VestingSchedule)> {
+        let remaining_amount = total_amount
+            .checked_sub(transferred_amount)
+            .ok_or(ErrorCode::GenericOverflow)?;
```

Comment:
You could even throw `SplitTooManyTokens` here

Reply:
Otherwise that error is not used

Reply:
the SplitTooManyTokens code implies that this method is only called in the split/accept methods. what happens if someone calls this somewhere else?